invenio_communities/communities/schema.py

```diff
@@ -240,31 +248,20 @@ def post_dump(self, data, many, **kwargs):
 
         return data
 
-    @pre_load
-    def initialize_custom_fields(self, data, **kwargs):
-        """Ensure custom fields are initialized.
-
-        We need to do that so that validation can take place in case a configured
-        field is marked as required.
-        """
-        data.setdefault("custom_fields", {})
-        return data
-
-    @post_load
-    def lowercase(self, in_data, **kwargs):
-        """Ensure slug is lowercase."""
-        in_data["slug"] = in_data["slug"].lower()
-        return in_data
-
 
 class CommunityParentSchema(BaseCommunitySchema):
     """Community parent schema."""
 
+    id = fields.String(required=True)
+    slug = SanitizedUnicode()
+    metadata = NestedAttribute(CommunityMetadataSchema)
+    access = NestedAttribute(CommunityAccessSchema)
```

Comment:
@slint need your review on this.

Reply:
But can't we make the `parent` field dump-only, and then using the `post_load` fish out the `parent.id` if necessary? Then there's no need to align e.g. specific fields that we load/dump.

Reply:
Hmm, if I am not mistaken, if we mark the parent as `dump_only` inside the `post_load` hook the parent field is empty , maybe @yashlamba can confirm this.

From what I see in the docs (see image below) we could do what you say but we would need to set the `unknown` option to `INCLUDE` (so that it's not empty on `post_load` and we can fish the id)
![Screenshot from 2024-02-15 10-14-24](https://github.com/inveniosoftware/invenio-communities/assets/15194802/fe9d49eb-8901-439e-b93f-18fe3644a422)

Reply:
`@post_load` can have access to the original data by passing `pass_original=True`, i.e.:


```python
    @post_load(pass_original=True)
    def load_parnet_id(self, data, original_data=None, **kwargs):
        """Load parent ID if present."""
        parent_id = original_data.get("parent", {}).get("id")
        if parent_id:
            data["parent"] = {"id": parent_id}
        return data
```

Reply:
Ah cool! that's way better :+1: